lib/submit-analytics.js

```diff
@@ -33,50 +104,184 @@ function sha256(data) {
+            };
+
+            resolve(payload);
+            return;
+        });
+    });
+}
+
+/**
+ * Generate and submit analytics data to servers.
+ * @param  {[webhookProdDetails|webhookStageDetails|mixpanelDetails]} requestFactory Submission URL generator
+ * @param  {Object} context contents of the app's package.json file
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns {Promise} Promise that resolves when analytics data has been submitted
+ */
+async function submitAnalytics(requestFactory, context, eventName) {
+    return new Promise((resolve, reject) => {
         setTimeout(() => {
             if (isAnalyticsDisabled()) {
+                // do nothing
+                resolve();
                 return;
             }
 
-            machineId()
-                .catch() // Ignore errors
-                .then((identifier) => {
-                    if (!identifier) {
-                        identifier = sha256('unknown');
-                    }
+            fetchPlatformData(context, eventName)
+            .catch((e) => {
+                reject("Failed to fetch platform data  [Error: " + e + "]");
+                return;
+            })
+            
+            .then((payload) => {
+                const versionTags = payload['mixPanel']['properties']['Version'].split('.');
 
-                    const payload = {
-                        'event': eventName,
-                        'properties': {
-                            'token': 'aab85907a13e1ff44a95be539d9942a9',
-                            'distinct_id': identifier,
-                            'Anonymized Machine Identifier': identifier,
-                            'Anonymized Application ID': sha256(__dirname),
-                            'Binding': context || 'node.js',
-                            'Version': require('../package.json').version,
-                            'Language': 'javascript',
-                            'OS Type': os.platform(),
-                            'OS Version': os.release(),
-                            'Node.js versions': process.versions
-                        }
-                    };
-
-                    request(`https://api.mixpanel.com/track/?data=${Buffer.from(JSON.stringify(payload), 'utf8').toString('base64')}&ip=1`,
-                    () => { /* Analytics failed. Do nothing. */ });
+                if (versionTags[0] < 10) {
+                    // on v6.x, we submit both MixPanel and S3 analytics
+                    // MixPanel analytics make no distinction between production and staging modes
+                    dispatchAnalytics(new mixpanelDetails(), payload);
+                }
+
+                dispatchAnalytics(requestFactory, payload)
+                .catch((e) => {
+                    reject(e);
+                    return;
+                })
+                .then(() => {
+                    resolve('');
```

Comment:
Little need to resolve to an empty string over `undefined` here.

```suggestion
                    resolve();
```